packages/framework/src/Model/Transport/TransportVisibilityCalculation.php

```diff
@@ -70,4 +73,21 @@ public function filterVisible(array $transports, array $visiblePaymentsOnDomain,
 
         return $visibleTransports;
     }
+
+    /**
+     * @param \Shopsys\FrameworkBundle\Model\Transport\Transport[] $transports
+     * @param \Shopsys\FrameworkBundle\Model\Cart\Cart $cart
+     * @return \Shopsys\FrameworkBundle\Model\Transport\Transport[]
+     */
+    public function filterTransportsByProductsInCart(array $transports, Cart $cart): array
+    {
+        $excludedTransports = array_merge(
+            ...array_map(static fn (Product $product) => $product->getExcludedTransports(), $cart->getProducts()),
```

Comment:
This loading of excluded transports in a loop is n+1 problem-prone.
we may basically achieve the desired behavior with a simple select
`Select distinct transport_id from products_excluded_transports where product_id in (...)`

then we can just drop transports with ids from returned query

Reply:
so what about this? 202254e3b8

I was kind of struggling with the right placement of the query as there is no entity `ProductExcludedTransport` hence was not convinced about creating the `ProductExcludedTransportRepository`  (and facade?) classes. Another idea was adding the method into `ProductRepository` (and facade? :smile: ), I could even use query builder but I would have to use  `SELECT FROM Product::class p JOIN p.excludedTransports ` which seemed overcomplicated so in the end, I just post the native query directly here where I needed it. I am open to discussion of course :slightly_smiling_face: